### PR TITLE
ECS: Fix, reable systems to auto delete + refactor systemManager + re…

### PR DESCRIPTION
**Pull request by guillaumeAbel**
…factor Registry custom sparseArrays

MINOR

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Norm of the code has been respected

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of change does this PR introduce?** (You can choose multiple)
- [x] Bug fix
- [ ] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **What is the current behavior?** (link an issue based on the kind of change this pr introduce)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: src/Client/SceneManager.cpp [hunk 1]
```diff
@@ -31,22 +26,6 @@ static void initRaylib()
     InitWindow(screenWidth, screenHeight, "R-Bus");
     SetTargetFPS(fps);
     InitAudioDevice();
-    Registry::getInstance().addEntity();
```
Comment (guillaumeAbel) [inline src/Client/SceneManager.cpp:67]: this deleted code is some poc, moved in a system for more poc things like proving that system can auto delete after creating player entity.

File: src/Client/SceneManager.cpp [hunk 2]
```diff
@@ -6,21 +6,16 @@
 */
 
 #include "SceneManager.hpp"
+#include <iostream>
```
Comment (romainpanno) [inline src/Client/SceneManager.cpp:9]: sys include on top

Comment (guillaumeAbel) [reply src/Client/SceneManager.cpp:9]: pas le choix c le formateur qui fait ça quand je les mets dans le bon ordre

File: src/Client/SceneManager.hpp [hunk 3]
```diff
@@ -13,17 +13,20 @@
...
 
-enum SystemManagers { GAME, EVENTS, DISPLAY, MANAGER_MAX };
+enum SystemManagers { GAME, EVENTS, DISPLAY };
 
 class SceneManager {
     public:
         static SceneManager &getInstance();
         int run();
         void changeScene(Scene scene);
+        Scene getCurrentScene();
```
Comment (kw4ntiq) [inline src/Client/SceneManager.hpp]: Je sais pas si la scene sera modifiée ailleurs mais dans le cas contraire faudrait mettre Scene getCurrentScene() const;

**Review by kw4ntiq — APPROVED** [review]

**Review by kw4ntiq — APPROVED** [review]

File: src/Client/Systems/Events/EventsSystems.cpp [hunk 4]
```diff
@@ -39,9 +41,22 @@ namespace Systems {
         }
     }
 
-    std::vector<std::function<void(std::size_t)>>
+    void
+    EventsSystems::changeScene(std::size_t /*unused*/, std::size_t /*unused*/)
+    {
+        if (IsKeyDown(KEY_J)) {
```
Comment (Saverio976) [inline src/Client/Systems/Events/EventsSystems.cpp]: Raylib::

**Review by Saverio976 — CHANGES_REQUESTED** [review]

File: src/Client/Systems/Events/EventsSystems.cpp (hunk 4 above)
Comment (guillaumeAbel) [inline src/Client/Systems/Events/EventsSystems.cpp]:
```suggestion
        if (IsKeyDown(Raylib::KEY_J)) {
```

**Review by Saverio976 — APPROVED** [review]

**Pull request merged**